app/assets/javascripts/repositories/repository_datatable.js

```diff
@@ -32,27 +50,429 @@ var RepositoryDatatable = (function(global) {
+        $('#unassignRepositoryRecords').removeClass('disabled');
+        $('#unassignRepositoryRecords').prop('disabled', false);
+      }
+    } else if (currentMode === 'editMode') {
+      $('#repository-acitons-dropdown').addClass('disabled');
+      $('#repository-acitons-dropdown').prop('disabled', true);
+      $('.dataTables_length select').prop('disabled', true);
+      $('#addRepositoryRecord').addClass('disabled');
+      $('#addRepositoryRecord').prop('disabled', true);
+      $('#editRepositoryRecord').addClass('disabled');
+      $('#editRepositoryRecord').prop('disabled', true);
+      $('#addNewColumn').addClass('disabled');
+      $('#addNewColumn').prop('disabled', true);
+      $('#deleteRepositoryRecordsButton').addClass('disabled');
+      $('#deleteRepositoryRecordsButton').prop('disabled', true);
+      $('#exportRepositoriesButton').off('click');
+      $('#export-repositories').off('click');
+      $('#assignRepositoryRecords').addClass('disabled');
+      $('#assignRepositoryRecords').prop('disabled', true);
+      $('#unassignRepositoryRecords').addClass('disabled');
+      $('#unassignRepositoryRecords').prop('disabled', true);
+      $('#repository-columns-dropdown').find('.dropdown-toggle').prop('disabled', true);
+      $('th').addClass('disable-click');
+      $('.repository-row-selector').addClass('disabled');
+      $('.repository-row-selector').prop('disabled', true);
+    }
+  }
+
+  // Updates "Select all" control in a data table
+  function updateDataTableSelectAllCtrl() {
+    var $table = TABLE.table().node();
+    var $header = TABLE.table().header();
+    var $chkboxAll = $('.repository-row-selector', $table);
+    var $chkboxChecked = $('.repository-row-selector:checked', $table);
+    var chkboxSelectAll = $('input[name="select_all"]', $header).get(0);
 
-  var originalHeader;
+    // If none of the checkboxes are checked
+    if ($chkboxChecked.length === 0) {
+      chkboxSelectAll.checked = false;
+      if ('indeterminate' in chkboxSelectAll) {
+        chkboxSelectAll.indeterminate = false;
+      }
 
-  // Tells whether to filter only assigned repository records
-  var viewAssigned;
+    // If all of the checkboxes are checked
+    } else if ($chkboxChecked.length === $chkboxAll.length) {
+      chkboxSelectAll.checked = true;
+      if ('indeterminate' in chkboxSelectAll) {
+        chkboxSelectAll.indeterminate = false;
+      }
+
+    // If some of the checkboxes are checked
+    } else {
+      chkboxSelectAll.checked = true;
+      if ('indeterminate' in chkboxSelectAll) {
+        chkboxSelectAll.indeterminate = true;
+      }
+    }
+  }
+
+  // Helper functions
+  function listItemDropdown(options, currentValue, columnId) {
+    var html = '<select class="form-control selectpicker repository-dropdown" ';
+    html += 'data-abs-min-length="2" data-live-search="true" ';
+    html += 'data-container="body" column_id="' + columnId + '">';
+    html += '<option value="-1"></option>';
```

Comment:
For multi-line strings we can use [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals), anyway we not support old IE browsers.